tests/unit/backend/chalice/api_server/test_v1_dataset.py

```diff
@@ -98,3 +98,107 @@ def test__minimal_status__ok(self):
             processing_status_updater(processing_status_id, processing_status)
             response = self.app.get(test_url.url, headers=headers)
             self.assertEqual(json.loads(response.body)["upload_status"], status.name)
+
+    def test__cancel_dataset_download__ok(self):
+        # Test pre upload
+        collection = self.generate_collection(visibility=CollectionVisibility.PRIVATE.name)
+        processing_status = {"upload_status": UploadStatus.WAITING, "upload_progress": 0.0}
+        dataset = self.generate_dataset(collection=collection, processing_status=processing_status)
+        test_url = f"/dp/v1/datasets/{dataset.id}"
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        response = self.app.delete(test_url, headers=headers)
+
+        self.assertEqual(response.status_code, 202)
+
+        # Test while uploading
+        processing_status = {"upload_status": UploadStatus.UPLOADING, "upload_progress": 10.0}
+        dataset = self.generate_dataset(collection=collection, processing_status=processing_status)
+        test_url = f"/dp/v1/datasets/{dataset.id}"
+
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        response = self.app.delete(test_url, headers=headers)
```

Comment:
What happens if you call this twice?

Reply:
It should be a 403 if we call this again. Can you add that to this test, unless it exists somewhere else?

Reply:
according to the ticket it should be a 202